force-app/main/default/lwc/hero/__tests__/hero.test.js

```diff
@@ -0,0 +1,186 @@
+import { createElement } from 'lwc';
+import Hero from 'c/hero';
+import IMAGE_URL from '@salesforce/resourceUrl/bike_assets';
+
+// Mock realistic data for the public properties
+const mockTitle = 'Title';
+const mockSlogan = 'Slogan';
+const mockButtonText = 'Click Me!';
+const mockHeroDetailsPositionLEFT = 'left';
+const mockHeroDetailsPositionRIGHT = 'right';
+const mockResourceUrl = 'www.salesforce.com';
+const mockImgOrVideoIMAGE = 'Image';
+const mockImgOrVideoVIDEO = 'Video';
+const mockInternalResource = true;
+const mockOverlay = 'true';
+const mockOpacity = 10;
+const mockButtonClickProductOrFamilyName = 'Product';
+
+describe('c-hero', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('displays an image in the hero when the type is image', () => {
+        const element = createElement('c-hero', {
+            is: Hero
+        });
+        element.title = mockTitle;
```

Comment:
General FYI: no need to set all attributes if you're not testing them. Most of these are optional so you could greatly reduce the amount of code in your tests. Please do that so we can clearly see what's being tested.

For example, in this test you would only need to set `imgOrVideo` and `resourceUrl`.